setup.py

```diff
@@ -27,14 +27,68 @@
     "rsa>=3.1.4,<5",
 )
 
+# TODO(https://github.com/googleapis/google-auth-library-python/issues/1440): Unit test fails with
+#  `No module named 'cryptography.hazmat.backends.openssl.x509' for Python 3.7``.
+cryptography_base_require = [
+    "cryptography>=38.0.3",
+    "cryptography < 39.0.0; python_version < '3.8'",
+]
+
+requests_extra_require = ["requests >= 2.20.0, < 3.0.0.dev0"]
+
+aiohttp_extra_require = ["aiohttp >= 3.6.2, < 4.0.0.dev0", *requests_extra_require]
+
+pyjwt_extra_require = ["pyjwt>=2.0", *cryptography_base_require]
+
+reauth_extra_require = ["pyu2f>=0.1.5"]
+
+enterprise_cert_extra_require = ["cryptography", "pyopenssl"]
```

Comment:
If there is a reason that we need to allow all versions of cryptography, can we add a comment even if it is just a link to a bug to follow up?

```suggestion
enterprise_cert_extra_require = [*cryptography_base_require, "pyopenssl"]
```

Reply:
Filed https://github.com/googleapis/google-auth-library-python/issues/1738.